.github/workflows/build.yml

```diff
@@ -54,13 +54,13 @@ jobs:
           flake8 --ignore=D,W503,W504 geomstats examples tests
           flake8 geomstats/geometry geomstats/learning 
 
-      - name: unit testing [nose2]
+      - name: unit testing [pytest]
         env: 
           GEOMSTATS_BACKEND : ${{matrix.geomstats-backend}} 
         run: |
-          nose2 --with-coverage ${{matrix.test-folder}}
+          pytest --cov-report term --cov=geomstats ${{matrix.test-folder}} 
```

Comment:
I'm confused here: does this mean that the coverage is only reported when running the `matrix.test-folder`, i.e. just tests/tests_geomstats or tests/test_notebooks? 

I know it was already like this in the former build.yml. But should we hardcode tests/test_geomstats as the folder here?